vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/ConsumableResource.java

```diff
@@ -77,8 +77,10 @@ public JavaType typeFromId(DatabindContext context, String id) throws IOExceptio
    * @param workflowName the name of the workflow
    * @param workflowVersion the version of the workflow
    * @param vidarrId the identifier of the workflow run
+   * @param isComplete a boolean flag specifying whether the run has completed
+   * @param input the input requested from the submitter, if applicable and provided.
    */
-  void release(String workflowName, String workflowVersion, String vidarrId);
+  void release(String workflowName, String workflowVersion, String vidarrId, boolean isComplete, Optional<JsonNode> input);
```

Comment:
taking a final look at this, isComplete and input are matched - in the 3 calls to release() in the server, input is always empty when isComplete is true, and there is input the one time isComplete is false. That means input.isPresent is functionally equivalent to isComplete. You could refactor this to drop isComplete, or always provide input